packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -16,6 +16,43 @@ import '../cache.dart';
 import '../globals.dart';
 import 'sdk.dart';
 
+/// Represents Flutter-specific data that is added to the `PUB_ENVIRONMENT`
+/// environment variable and allows understanding the type of requests made to
+/// the package site on Flutter's behalf.
+// DO NOT update without contacting kevmoo.
+// We have server-side tooling that assumes the values are consistent.
+class PubContext {
+  static final RegExp _validContext = new RegExp('[a-z][a-z_]*[a-z]');
+
+  static final PubContext create = new PubContext._(<String>['create']);
+  static final PubContext createPackage = new PubContext._(<String>['create_pkg']);
+  static final PubContext createPlugin = new PubContext._(<String>['create_plugin']);
+  static final PubContext interactive = new PubContext._(<String>['interactive']);
+  static final PubContext pubGet = new PubContext._(<String>['get']);
+  static final PubContext pubUpgrade = new PubContext._(<String>['upgrade']);
+  static final PubContext runTest = new PubContext._(<String>['run_test']);
+
+  static final PubContext flutterTests = new PubContext._(<String>['flutter_tests']);
+  static final PubContext updatePackages = new PubContext._(<String>['update_packages']);
+
+  final List<String> _values;
+
+  PubContext._(this._values) {
```

Comment:
style nits: please follow flutter style, which has constructors first, avoids newlines in expressions containing `=>`, and always has a newline before the `)` if there's one after the `(`. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo